src/queue.c

```diff
@@ -7013,6 +7013,7 @@ _dispatch_sig_thread(void *ctxt DISPATCH_UNUSED)
 	_dispatch_clear_stack(0);
 #if defined(_WIN32)
 	Sleep(INFINITE);
+	__builtin_unreachable();
```

Comment:
I think that I prefer the `__assume(0)` spelling just to keep hope for someday being able to use MSVC again.

Reply:
`__builtin_unreachable()` is consistent with the spelling elsewhere in the repo. Ideally, we'd replace all uses of it with something like `dispatch_unreachable()` that resolves to the correct intrinsic under both compilers. But that's beyond the scope of this PR.